x/crosschain/keeper/cctx_utils_outbound.go

```diff
@@ -0,0 +1,196 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	cosmoserrors "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k Keeper) GetOutbound(ctx sdk.Context, cctx *types.CrossChainTx, msg types.MsgVoteOnObservedOutboundTx, ballotStatus observertypes.BallotStatus) error {
+	if ballotStatus != observertypes.BallotStatus_BallotFinalized_FailureObservation {
+		if !msg.ValueReceived.Equal(cctx.GetCurrentOutTxParam().Amount) {
+			ctx.Logger().Error(fmt.Sprintf("VoteOnObservedOutboundTx: Mint mismatch: %s value received vs %s cctx amount",
+				msg.ValueReceived,
+				cctx.GetCurrentOutTxParam().Amount))
+			return cosmoserrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("ValueReceived %s does not match sent value %s", msg.ValueReceived, cctx.GetCurrentOutTxParam().Amount))
+		}
+	}
+	// Update CCTX values
+	cctx.GetCurrentOutTxParam().OutboundTxHash = msg.ObservedOutTxHash
+	cctx.GetCurrentOutTxParam().OutboundTxGasUsed = msg.ObservedOutTxGasUsed
+	cctx.GetCurrentOutTxParam().OutboundTxEffectiveGasPrice = msg.ObservedOutTxEffectiveGasPrice
+	cctx.GetCurrentOutTxParam().OutboundTxEffectiveGasLimit = msg.ObservedOutTxEffectiveGasLimit
+	cctx.GetCurrentOutTxParam().OutboundTxObservedExternalHeight = msg.ObservedOutTxBlockHeight
+	cctx.CctxStatus.LastUpdateTimestamp = ctx.BlockHeader().Time.Unix()
+
+	tss, found := k.zetaObserverKeeper.GetTSS(ctx)
+	if !found {
+		return types.ErrCannotFindTSSKeys
+	}
+	if tss.TssPubkey != cctx.GetCurrentOutTxParam().TssPubkey {
+		return types.ErrTssMismatch
+	}
+	return nil
+}
+
+func (k Keeper) FundStabiltityPool(ctx sdk.Context, cctx *types.CrossChainTx) {
+	// Fund the gas stability pool with the remaining funds
+	if err := k.FundGasStabilityPoolFromRemainingFees(ctx, *cctx.GetCurrentOutTxParam(), cctx.GetCurrentOutTxParam().ReceiverChainId); err != nil {
```

Comment:
Is `msg.OutTxChain` (used in previous version) equal to `cctx.GetCurrentOutTxParam().ReceiverChainId)` in all cases.

Reply:
the pending nonces is set using the receiver chain ID 
https://github.com/zeta-chain/zeta-node/blob/75580e7726f421b40be995f3a17de1cce28e338a/x/crosschain/keeper/cctx_utils.go#L105-L118

and the client uses that to query and process outbounds .
https://github.com/zeta-chain/zeta-node/blob/75580e7726f421b40be995f3a17de1cce28e338a/zetaclient/zetacore_observer.go#L145-L149 ,